rusk/src/lib/builder/http_only.rs

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use tokio::sync::broadcast;
+use tracing::info;
+
+use crate::http::{DataSources, HttpServer, HttpServerConfig};
+
+#[derive(Default)]
+pub struct RuskHttpBuilder {
+    http: Option<HttpServerConfig>,
+}
+
+impl RuskHttpBuilder {
+    pub fn with_http(mut self, http: HttpServerConfig) -> Self {
+        self.http = Some(http);
+        self
+    }
+
+    pub async fn build_and_run(self) -> anyhow::Result<()> {
+        let (_rues_sender, rues_receiver) = broadcast::channel(1);
+
+        let mut _ws_server = None;
+        if let Some(http) = self.http {
+            info!("Configuring HTTP");
+
+            #[allow(unused_mut)]
+            let mut handler = DataSources::default();
```

Comment:
Any reason why Datasource is not passed as a parameter? It might be handy to extend a testing framework

Reply:
Opened #2453 to potentially address this